### PR TITLE
[WEJBHTTP-151] Refactor JNDI server side

### PR DESCRIPTION
**Pull request by ropalka**
https://issues.redhat.com/browse/WEJBHTTP-151

### CODE REVIEW

File: naming/src/main/java/org/wildfly/httpclient/naming/HttpRemoteNamingService.java [hunk 1]
```diff
@@ -131,7 +131,15 @@ public final void handleRequest(HttpServerExchange exchange) throws Exception {
                 } else if (result instanceof Context) {
                     exchange.setStatusCode(StatusCodes.NO_CONTENT);
                 } else {
-                    doMarshall(exchange, result);
```
Comment (rachmatowicz) [inline naming/src/main/java/org/wildfly/httpclient/naming/HttpRemoteNamingService.java:134]: Are you eliminating this method as it is only ever called once?

File: naming/src/main/java/org/wildfly/httpclient/naming/HttpRemoteNamingService.java [hunk 2]
```diff
@@ -101,21 +101,34 @@ private void registerHandler(final RoutingHandler routingHandler, final RequestT
...
-            case CREATE_SUBCONTEXT: return new CreateSubContextHandler();
-            case DESTROY_SUBCONTEXT: return new DestroySubcontextHandler();
-            case LIST: return new ListHandler();
-            case LIST_BINDINGS: return new ListBindingsHandler();
-            case LOOKUP: return new LookupHandler();
-            case LOOKUP_LINK: return new LookupLinkHandler();
-            case REBIND: return new RebindHandler();
-            case RENAME: return new RenameHandler();
-            case UNBIND: return new UnbindHandler();
+            case BIND: return new BindHandler(localContext, httpServiceConfig, classResolverFilter);
```
Comment (rachmatowicz) [inline naming/src/main/java/org/wildfly/httpclient/naming/HttpRemoteNamingService.java]: What is the motivation for making these static?  I can see you have also parameterized the constructors for eventual use in the ServerHandlers factory.

**Review by rachmatowicz — COMMENTED** [review]
Looks good overall.

File: naming/src/main/java/org/wildfly/httpclient/naming/HttpRemoteNamingService.java (hunk 2 above)
Comment (ropalka) [reply naming/src/main/java/org/wildfly/httpclient/naming/HttpRemoteNamingService.java]: I made them static to be independent of HttpRemoteNamingService class - preparation step to extracting server handlers out from this class @rachmatowicz .

**Pull request merged**